### DISCUSSION TITLE
ily

### DISCUSSION THREAD

**Discussion by spookyuser** [craigwh10/dependency_install_swapper#31] [General]
<3

**Comment by spookyuser** [comment]
What about if this listened to the clipboard and just replaced npm with pnpm?

**Comment by craigwh10** [comment]
Hey thanks for reaching out @spookyuser, could you elaborate a bit on "listened to the clipboard" - do you mean some form of a CLI plugin that listens to when you've pasted from the clipboard an npm install command and makes it automatically change to the package manager that is in use within the repository?

**Comment by spookyuser** [comment]
not really idk if this is possible but i don't want this just to work on the registeries (like npm.io, etc) i want it to work on anywebsite i copy the text "npm ...." and replace that with pnpm, 	but i'm not sure if there's even an api for that

**Comment by craigwh10** [comment]
I understand now, could be possible yeah but would need to prove a few bits out with some proper digging https://medium.com/front-end-weekly/how-to-listen-to-the-contents-of-the-clipboard-in-the-browser-aadb0accaa19 which shows something similar to what you said, listening from the extension via copy/paste events if it contains the command string, notifying the user in some fashion that the clipboard can be replaced with preferred (rather than just replacing as that might not be great UX) - and once some confirmation has happened then replacing the clipboard with the preferred command.

Could be possible, neat idea.

**Reply by craigwh10** [reply]
Again would need to prove something out and I'm fairly rusty with extensions (been focusing more backend bits recently) - but a guestimate could be something like a content script that gets injected into each page, which emits the clipboard for a window from which a background/service worker would grab the state for and possibly update (somehow)

**Reply by craigwh10** [reply]
OR if not the clipboard, could make the updated command statefully available via a text field that could be copied within the extension popup and pasted which follows a similar process as above

**Reply by spookyuser** [reply]
great idea!

**Reply by spookyuser** [reply]
Or what about a native app that does have access to the clipboard and maybe notifies the extension when clipboard copies occur https://developer.chrome.com/docs/extensions/mv3/nativeMessaging/ although i guess at that point a native app will prob just be easier lol

**Comment by spookyuser** [comment]
oh another thing i would love is replacing `npm install -g` commands with `pnpx`
